TileFormats/VectorData/README.md

```diff
@@ -1,19 +1,248 @@
+    
+    uBuffer[i] = zigZag(u - lastU);
+    vBuffer[i] = zigZag(v - lastV);
+    heightBuffer = zigZag(height - lastHeight);
+
+    lastU = u;
+    lastV = v;
+    lastHeight = height;
+}
+```
+
+Example decoding code is listed below:
+```javascript
+function zigZagDecode(value) {
+    return (value >> 1) ^ (-(value & 1));
+}
+
+var u = 0;
+var v = 0;
+var height = 0;
+
+for (var i = 0; i < length; ++i) {
+    u += zigZagDecode(uBuffer[i]);
+    v += zigZagDecode(vBuffer[i]);
+    height += zigZagDecode(heightBuffer[i]);
+    
+    uBuffer[i] = u;
+    vBuffer[i] = v;
+    heightBuffer[i] = height;
+}
+```
+
+#### Polygon positions
+
+**Figure 3**: Polygon layout.
+
+![](figures/polygon.jpg)
+
+The number of positions for each polygon is determined by the value of its `POLYGON_COUNT`. For example, let the first polygon count be 5. The first polygons u values start at offset `0` and end at `5 * byteSize`. Its v values start at `polygonPositionsByteLength / 2` and end at `polygonPositionsByteLength / 2 + 5 * byteSize`.
+
+The positions of the polygons must be the outer ring positions listed in counter-clockwise order.
+
+#### Polyline positions
+
+**Figure 4**: Polyline layout.
+
+![](figures/polyline.jpg)
+
+The number of positions for each polyline is determined by the value of its `POLYLINE_COUNT`. From the first point on the polyline, each successive point creates a segment connected to the previous.
+
+#### Point positions
+
+**Figure 5**: Point layout.
+
+![](figures/point.jpg)
+
+Each `u, v, height` triple is a single point.
+
+### Meshes
+
+**Figure 6**: Mesh layout
+
+![](figures/mesh.jpg)
+
+The indices are a buffer of `uint32` values. The indices for each mesh must be contiguous in the buffer. For example, the first mesh would have indices starting at the byte offset given by the first element in `MESH_INDEX_OFFSETS` and end after the first element in `MESH_INDEX_COUNTS`. After the indices is a `float32[]` containing positions of tall of the meshes. The number of positions in the buffer is given by `MESH_POSITION_COUNT`. Unlike polygons, there are no restrictions on where a mesh can index into the position buffer.
+
+## File Extension
+
+`.vctr`
+
+The file extension is optional. Valid implementations ignore it and identify a content's format by the `magic` field in its header.
+
+## MIME Type
+
+_TODO, [#60](https://github.com/AnalyticalGraphicsInc/3d-tiles/issues/60)_
+
+`application/octet-stream`
+
+## Implementation Notes
+
+In JavaScript, a `TypedArray` cannot be created on data unless it is byte-aligned to the data type.
+For example, a `Float32Array` must be stored in memory such that its data begins on a byte multiple of four since each `float` contains four bytes.
+
+The string generated from the JSON and each array should be padded is necessary to ensure byte alignment.
+ 
+## Resources
+1. [quantized-mesh-1.0 terrain format](https://cesiumjs.org/data-and-assets/terrain/formats/quantized-mesh-1.0.html)
```

Comment:
I may have missed it, where is this referenced from?

Reply:
It isn't referenced anywhere. The spec for the position layout is similar. @pjcozzi, should I remove this?

Reply:
Up to you.